### ISSUE TITLE
Supervised Loss

### ISSUE THREAD

**Issue by yeefantan** [jsyoon0823/TimeGAN#69]
Hi Yoon, 

Thanks for the very interesting work!

I have a small concern regarding the supervised loss

As reported in the paper, supervised loss is used to capture the discrepancy between the real latent space captured by Embedder 
>H = embedder(X, T)

and also the synthetic next-step vector from generator 
>E_hat = generator(Z, T) 
>H_hat = supervisor(E_hat, T).

In the supervised loss, it is 
>G_loss_S = tf.losses.mean_squared_error(H[:,1:,:], H_hat_supervise[:,:-1,:])

However **H_hat_supervise** is defined as 
>H_hat_supervise = supervisor(H, T),

where **H_hat_supervise**  has no relations to the generator, and the **G_loss_S** seems like minimizing the discrepancy captured solely by the Embedder?

**Comment by jsyoon0823** [comment]
Thanks for your interest in our paper and code.

In this code, you can think of the combination of generator and supervisor
as the generator in the paper.
You can also see that the synthetic data X_hat depends on both generator
and supervisor.

Thanks!

**Comment by yeefantan** [comment]
Got it, thanks!
